### PR TITLE
Gitness App : repo commits page

### PR DESCRIPTION
**Pull request by shaurya-harness**
Gitness App : repo commits page

added commits list page at repo level
TODO : branch selection should fetch new commits as per the new branch

### CODE REVIEW

File: apps/gitness/src/pages/repo/repo-commits.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    repo_ref: repoRef,
+    queryParams: { page: 0, limit: 10 }
+  })
+
+  const renderContent = () => {
+    if (isFetchingCommits) {
+      return <SkeletonList />
+    }
+    // @ts-expect-error remove "@ts-expect-error" once type issue for "content" is resolved
+    if (commitData?.content?.commits?.length) {
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/repo/repo-commits.tsx:38]: `.content` and the `ts-expect-error` are not needed anymore

File: apps/gitness/src/pages/repo/repo-commits.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        <ListActions.Right>
+          <ListActions.Dropdown title="Filter" items={filterOptions} />
+          <ListActions.Dropdown title="Sort" items={sortOptions} />
+        </ListActions.Right>
+      </ListActions.Root>
+      <Spacer size={5} />
+      {renderContent()}
+      <Spacer size={8} />
+      {!isFetchingCommits && (
+        <ListPagination.Root>
```
Comment (abhinavrastogi-harness) [inline apps/gitness/src/pages/repo/repo-commits.tsx:86]: we can remove dummy stuff from gitness app.. if pagination is not implemented/supported, we should not add it

File: apps/gitness/src/pages/repo/repo-commits.tsx (hunk 1 above)
Comment (shaurya-harness) [reply apps/gitness/src/pages/repo/repo-commits.tsx:38]: that PR by @vardanbansal-harness is not merged yet

File: apps/gitness/src/pages/repo/repo-commits.tsx (hunk 2 above)
Comment (shaurya-harness) [reply apps/gitness/src/pages/repo/repo-commits.tsx:86]: yeah makes sense

**Review by vardanbansal-harness — APPROVED** [review]

File: apps/gitness/src/pages/repo/repo-commits.tsx (hunk 1 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/pages/repo/repo-commits.tsx:26]: I will remove this as well once my PR gets merged.

**Pull request merged**